tests/unit/Codeception/Test/GherkinTest.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+class GherkinTest extends PHPUnit_Framework_TestCase
+{
+
+    protected $feature;
+    public static $calls = '';
+
+    /**
+     * @var \Codeception\Test\Loader\Gherkin
+     */
+    protected $loader;
+
+    protected function setUp()
+    {
+        $this->loader = new \Codeception\Test\Loader\Gherkin(
+            [
+                'gherkin' => [
+                    'contexts' => [
+                        'default' => ['GherkinTestContext']
+
+                    ]
+                ]
+            ]
+        );
+        self::$calls = '';
+    }
+
+    protected function getServices()
+    {
+        return [
+            'di'         => new \Codeception\Lib\Di(),
+            'dispatcher' => new \Codeception\Util\Maybe(),
+            'modules'    => \Codeception\Util\Stub::makeEmpty('Codeception\Lib\ModuleContainer')
+        ];
+    }
+
+    public function testLoadGherkin()
+    {
+        $this->loader->loadTests(codecept_data_dir('refund.feature'));
+        $tests = $this->loader->getTests();
+        $this->assertCount(1, $tests);
+        /** @var $test \Codeception\Test\Gherkin  * */
+        $test = $tests[0];
+        $this->assertInstanceOf('\Codeception\Test\Gherkin', $test);
+        $this->assertEquals('Jeff returns a faulty microwave', $test->getFeature());
+    }
+
+    /**
+     * @depends testLoadGherkin
+     */
+    public function testLoadWithContexts()
+    {
+        $this->loader->loadTests(codecept_data_dir('refund.feature'));
+        $test = $this->loader->getTests()[0];
+        /** @var $test \Codeception\Test\Gherkin  * */
+        $test->getMetadata()->setServices($this->getServices());
+        $test->test();
+        $this->assertEquals('abc', self::$calls);
+    }
+
+    public function testTags()
+    {
+        $this->loader = new \Codeception\Test\Loader\Gherkin(
+            [
+                'gherkin' => [
+                    'contexts' => [
+                        'default' => ['GherkinTestContext'],
+                        'tag'     => [
+                            'important' => ['TagGherkinContext']
+                        ]
+                    ]
+                ]
+            ]
+        );
+        $this->loader->loadTests(codecept_data_dir('refund.feature'));
+        $test = $this->loader->getTests()[0];
+        /** @var $test \Codeception\Test\Gherkin  * */
+        $test->getMetadata()->setServices($this->getServices());
+        $test->test();
+        $this->assertEquals('aXc', self::$calls);
+    }
+
+    public function testMatchingPatterns()
+    {
+        $pattern = 'hello :name, are you from :place?';
+        $regex = $this->loader->makePlaceholderPattern($pattern);
+        $this->assertEquals('/hello "(.*?)", are you from "(.*?)"\?/', $regex);
```

Comment:
there are 2 step formats: regular string with placeholders and regex
so yes, we convert placeholders to a compatible regex.
In regex string we use quotes to highlight the variables.